File: Arrays/getPairsCount.py
def getPairsCount(arr, sum):
    count = 0
    viewed = [0] * 150
    for val in arr:
        viewed[val] += 1
    
    for val in arr:
        count += viewed[sum - val]

        if(sum - val == val):
            count -= 1
        
    return count//2

def getPairsCountBrute(arr, sum):
    count = 0
    for i in range(len(arr)):
        for j in range(i+1, len(arr)):
            if arr[i] + arr[j] == sum:
                count += 1
    return count

arr = [1, 5, 7, -1, 5]
sum = 6
# pairs = getPairsCount(arr, sum)
pairs = getPairsCountBrute(arr, sum)
print("There are {} pairs in the array with the sum {}".format(pairs, sum))

File: cpp/arrays/getPairsCount.cpp
#include <bits/stdc++.h>
using namespace std;

int getPairsCount(int arr[], int n, int val){
    int count = 0;
    int viewed[150] = {0};
    
    for(int i=0; i<n; i++){
        viewed[arr[i]]++;
    }
    
    for(int i=0; i<n; i++){
        count += viewed[val - arr[i]];
        if(val - arr[i] == arr[i])
            count--;
    }
    return count/2;
}

int getPairsCountBrute(int arr[], int n, int val){
    int count = 0;
    for(int i=0; i<n; i++){
        for(int j=i+1; j<n; j++){
            if(arr[i] + arr[j] == val)
                count++;
        }
    }
    return count;
}

int main(){
    int arr[] = {1, 5, 7, -1, 5};
    int n = sizeof(arr)/ sizeof(arr[0]);
    int val = 6;
    
    int pairs = getPairsCount(arr, n, val);
    // int pairs = getPairsCountBrute(arr, n, val);
    cout << "There are " << pairs << " pairs in the array with the sum " << val;
}